src/openrct2-ui/drawing/engines/opengl/DrawCommands.h

```diff
@@ -21,6 +21,49 @@
 #include "GLSLTypes.h"
 #include "TextureCache.h"
 
+
+template<typename T>
+class CommandBatch
```

Comment:
This whole class really looks and feels like `vector`, is it really needed?

Reply:
We never use insert/remove, and I would need per batch the index.

Reply:
Calling allocates guarantees a reference allowing you to write directly into the vector storage. It only resizes when required and therefor we have zero copies.